pom.xml

```diff
@@ -54,27 +51,24 @@
                 </configuration>
             </plugin>
             <plugin>
-                <groupId>org.codehaus.mojo</groupId>
-                <artifactId>exec-maven-plugin</artifactId>
-                <version>1.6.0</version>
-                <executions>
-                    <execution>
-                        <id>git-version</id>
-                        <phase>generate-resources</phase>
-                        <goals>
-                            <goal>exec</goal>
-                        </goals>
-                    </execution>
-                </executions>
+                <groupId>pl.project13.maven</groupId>
+                <artifactId>git-commit-id-plugin</artifactId>
+                <version>4.0.2</version>
                 <configuration>
-                    <executable>sh</executable>
-                    <outputFile>target/generated-resources/pro/javacard/gp/pro_version.txt</outputFile>
-                    <arguments>
-                        <argument>-c</argument>
-                        <argument>[ "x$GPPRO_VERSION" = "x" ] &amp;&amp; git describe --tags --always --long --dirty ||
-                            echo "$GPPRO_VERSION"
-                        </argument>
-                    </arguments>
+                    <runOnlyOnce>false</runOnlyOnce>
+                    <generateGitPropertiesFile>true</generateGitPropertiesFile>
+                    <skipPoms>true</skipPoms>
+                    <generateGitPropertiesFilename>${project.build.directory}/generated-resources/pro/javacard/gp/git.properties</generateGitPropertiesFilename>
+                    <includeOnlyProperties>
```

Comment:
What are the other properties? If the file is not in tens of kilobytes, maybe it is easier to avoid these filters here.

Reply:
I thought it unnecessary to bundle committer email and whatnot since they're never read.